etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/utils/security/HashHelper.java

```diff
@@ -0,0 +1,24 @@
+package gov.hhs.cdc.trustedintermediary.etor.utils.security;
+
+import gov.hhs.cdc.trustedintermediary.wrappers.Logger;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import javax.inject.Inject;
+import org.apache.commons.codec.binary.Hex;
+
+public class HashHelper implements SecureHash {
+    @Inject Logger logger;
+
+    public <T> String generateHash(T input) {
+        try {
+            MessageDigest digest = MessageDigest.getInstance("SHA3-512");
+            byte[] objBytes = input.toString().getBytes(StandardCharsets.UTF_8);
+            byte[] hashBytes = digest.digest(objBytes);
+            return Hex.encodeHexString(hashBytes);
```

Comment:
I believe you don't need a third-party library.  Try `HexFormat` which is built into Java.

`return HexFormat.of().formatHex(hashBytes);`

Caveat, I've never used `HexFormat` before.

Reply:
We swapped out the third-party library.